src/platform/audio/alaudiomanager.cpp

```diff
@@ -0,0 +1,1440 @@
+
+        if (pap->openal.source == 0) {
+            Release_Playing_Audio(pap);
+            return nullptr;
+        }
+
+        m_binkPlayingAudio = pap;
+    }
+
+    return reinterpret_cast<void *>(m_binkPlayingAudio->openal.source);
+}
+
+/**
+ * Releases the audio handle for use by the video player.
+ */
+void ALAudioManager::Release_Bink_Handle()
+{
+    if (m_binkPlayingAudio != nullptr) {
+        Release_Playing_Audio(m_binkPlayingAudio);
+        m_binkPlayingAudio = nullptr;
+    }
+}
+
+/**
+ * Forces an audio event to be played.
+ */
+void ALAudioManager::friend_Force_Play_Audio_Event(AudioEventRTS *event)
+{
+    if (event->Get_Event_Info() == nullptr) {
+        Get_Info_For_Audio_Event(event);
+        if (event->Get_Event_Info() == nullptr) {
+            captainslog_warn("No info for forced audio event '%s'", event->Get_Event_Name().Str());
+            return;
+        }
+    }
+
+    // TODO:
+}
+
+/**
+ * Processes the request list for audio waiting to be played.
+ */
+void ALAudioManager::Process_Request_List()
+{
+    for (auto it = m_audioRequestList.begin(); it != m_audioRequestList.end();) {
+        if (*it != nullptr) {
+            if (Process_Request_This_Frame(*it)) {
+                if (!(*it)->m_isProcessed || Check_For_Sample(*it)) {
+                    Process_Request(*it);
+                }
+
+                (*it)->Delete_Instance();
+                it = m_audioRequestList.erase(it);
+            } else {
+                Adjust_Request(*it);
+                ++it;
+            }
+        }
+    }
+}
+
+/**
+ * Check if a request should be processed this frame.
+ *
+ * Inlined.
+ */
+bool ALAudioManager::Process_Request_This_Frame(AudioRequest *request)
+{
+    if (request->m_isAdding) {
+        return request->m_event.object->Get_Delay() < MSEC_PER_LOGICFRAME_REAL;
+    }
+
+    return true;
+}
+
+/**
+ * Reduced the delay until this request should be played by one frame.
+ *
+ * Inlined.
+ */
+void ALAudioManager::Adjust_Request(AudioRequest *request)
+{
+    if (request->m_isAdding) {
+        request->m_event.object->Decrement_Delay(MSEC_PER_LOGICFRAME_REAL);
+        request->m_isProcessed = true;
+    }
+}
+
+/**
+ * Checks the status of a request to determine if it should be processed.
+ *
+ * Inlined.
+ */
+bool ALAudioManager::Check_For_Sample(AudioRequest *request)
+{
+    if (request->m_isAdding) {
+        return true;
+    }
+
+    // If the object doesn't have info, try and retrieve it.
+    if (request->m_event.object->Get_Event_Info() == nullptr) {
+        Get_Info_For_Audio_Event(request->m_event.object);
+    }
+
+    if (request->m_event.object->Get_Event_Info()->Get_Visibility() == VISIBILITY_WORLD) {
+        return m_soundManager->Can_Play_Now(request->m_event.object);
+    }
+
+    return true;
+}
+
+/**
+ * Plays an audio event.
+ */
+void ALAudioManager::Play_Audio_Event(AudioEventRTS *event)
+{
+    const AudioEventInfo *aud_info = event->Get_Event_Info();
```

Comment:
Are all the function bodies just copies of milesaudiomanager, minus miles specific things? Or is there anything AL specific here already?

Reply:
There are a few AL specific things already. Much of the logic could probably be moved into the base class though